compare_platforms/compare_platforms_read_only.c

```diff
@@ -0,0 +1,193 @@
+/**
+ * This version can be run in both a CHERI and a non-CHERI environment to
+ * compare the two.
+ *
+ * In a CHERI environment, it will:
+ *     Step 1: Use can_read to check if the struct's "permissions" member includes write
+ * permissions. Step 2: Use capabilities to enforce read-only permissions (see
+ * https://github.com/capablevms/cheri-examples/blob/master/employee/include/employee.h). Step 3:
+ * Either: a) if command-line flag "-overwrite" was passed in, try to write to the read-only struct
+ * members, and SIGPROT; or b) try to exceed the bounds of an object and add write permissions
+ * (which will also SIGPROT).
+ *
+ * In a non-CHERI environment, it will:
+ *     Step 1: Use can_read to check if the struct's "permissions" member includes write
+ * permissions. Step 2: Try to exceed the bounds of an object and add write permissions (see
+ * https://ctsrd-cheri.github.io/cheri-exercises/exercises/buffer-overflow-heap/index.html). Step 3:
+ * Rewrite the review.
+ */
```

Comment:
It seems this formatting got broken, perhaps due to the `clang-format` call. Can you please try to find a way to redo it?

Reply:
Binary fixed in 35d8ec4d941d9833b54920d07b9b530f76f97afe

Top comment formatting fixed (and content corrected to reflect a change to the program flow) in 738e5ca2d1b374e09fb29b537cd28a75aded7e3e